src/clr-angular/data/datagrid/datagrid-page-size.spec.ts

```diff
@@ -0,0 +1,89 @@
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Component } from '@angular/core';
+
+import { ClrDatagridPageSize } from './datagrid-page-size';
+import { TestContext } from './helpers.spec';
+import { Page } from './providers/page';
+import { StateDebouncer } from './providers/state-debouncer.provider';
+
+export default function(): void {
+  describe('ClrDatagridPageSize component', function() {
+    describe('Template API', function() {
+      let context: TestContext<ClrDatagridPageSize, FullTest>;
+
+      beforeEach(function() {
+        context = this.create(ClrDatagridPageSize, FullTest, [Page, StateDebouncer]);
+      });
+
+      it('receives an input for page size options', function() {
+        context.testComponent.pageSizeOptions = [10, 20, 50, 100];
+        context.detectChanges();
+        expect(context.clarityDirective.pageSizeOptions).toEqual([10, 20, 50, 100]);
+      });
+    });
+
+    describe('View', function() {
+      describe('Default View Without Input Test', function() {
+        let context: TestContext<ClrDatagridPageSize, FullTest>;
+
+        beforeEach(function() {
+          context = this.create(ClrDatagridPageSize, SimpleTest, [Page, StateDebouncer]);
+        });
+
+        it('displays a select with a default pageSize if no input is given', function() {
+          const select = context.clarityElement.querySelector('select');
+          expect(select).not.toBeNull();
+          expect(select.innerText).toMatch('0');
+        });
+      });
+
+      describe('Full Test', function() {
+        let context: TestContext<ClrDatagridPageSize, FullTest>;
+        let pageProvider: Page;
+
+        beforeEach(function() {
+          context = this.create(ClrDatagridPageSize, FullTest, [Page, StateDebouncer]);
+          pageProvider = context.getClarityProvider(Page);
+        });
+
+        it('projects content before the select element', function() {
+          expect(context.clarityElement.textContent.trim()).toMatch('Hello world');
+        });
+
+        it('displays a select with pageSizeOptions as choices', function() {
+          context.testComponent.pageSizeOptions = [10, 20, 50, 100];
+          context.detectChanges();
+          const select = context.clarityElement.querySelector('select');
+          expect(select).not.toBeNull();
+          expect(select.innerText).toMatch('10\n20\n50\n100');
+        });
+
+        it('updates the page size upon pageSizeOption selection', function() {
+          pageProvider.size = 10;
+          pageProvider.totalItems = 100;
+          pageProvider.current = 1;
+          context.testComponent.pageSizeOptions = [10, 20, 50, 100];
+          context.detectChanges();
+
+          const select = context.clarityElement.querySelector('select');
+          expect(select).not.toBeNull();
+          select.value = '2: 20'; // setting it to the second option, with the expected value syntax when we use [ngValue]
```

Comment:
That's a bit brittle, using the `NgControl` or something would maybe work better, but don't worry about it. Just a general note, this is fine right now.